components/azure/azure.go

```diff
@@ -38,22 +39,18 @@ type ListResponse struct {
 }
 
 type CreateClusterAzure struct {
-	Node *CreateAzureNode `json:"node,omitempty"`
+	ResourceGroup     string                `json:"resourceGroup"`
+	KubernetesVersion string                `json:"kubernetesVersion"`
+	NodePools         *map[string]*NodePool `json:"nodePools,omitempty"`
 }
 
-type CreateAzureNode struct {
-	ResourceGroup     string `json:"resourceGroup"`
-	AgentCount        int    `json:"agentCount"`
-	AgentName         string `json:"agentName"`
-	KubernetesVersion string `json:"kubernetesVersion"`
+type NodePool struct {
+	AgentCount int    `json:"agentCount"`
+	VmSize     string `json:"vmSize"`
```

Comment:
This is a bit misleading. 
`nodePools: {
    "pool1" : {
        "agentCount":  1,
        "vmSize": "blah"
    }
}
`

We shouldn't mix agent and node pool terminology.